system/lib/libc/musl/include/signal.h

```diff
@@ -259,7 +259,7 @@ int sigandset(sigset_t *, const sigset_t *, const sigset_t *);
 
 #define SIG_ERR  ((void (*)(int))-1)
 #define SIG_DFL  ((void (*)(int)) 0)
-#define SIG_IGN  ((void (*)(int)) 1)
+#define SIG_IGN  ((void (*)(int))-2) /* XXX EMSCRIPTEN: use -2 since 1 is a valid function address */
```

Comment:
This change seems to be a regression for Python. Some code somewhere must assume that `SIG_IGN` is 1.

Reply:
Yeah the rational for the change is clear. I am trying to hunt down what is going on. The error reproduces in our CI so it's definitely from a clean build. But we could patch it back in Pyodide if need be.

@tiran Have you run into this problem?

Reply:
As a last resort we could shift the range of valid functions pointers from a range that starts at 1 to range that starts at something higher, such as 16, but it seems like quite in intrusive change (and not 100% free at runtime)... so I would much rather find an fix offending code.

Reply:
It looks to me like the Python signalmodule code has a bug that is fixed in v3.11: in v3.10 in compares the memory addresses of the Python objects, and it's returning False because they are two different `-2`'s. I suspect the Python interpreter interns 1 in some special way which makes the wrong code work by chance when the value in question is 1.

Reply:
https://github.com/python/cpython/blob/68f6a5da199826fef1098f869d8d2260aaccff8e/Modules/signalmodule.c#L552

Reply:
Okay I will apply the patch in @tiran's comment there.

Reply:
Thanks @sbc100

Reply:
Yeah, I ran into the problem, too. PR https://github.com/python/cpython/pull/31759 fixed the issue in 3.11 / main. I did not backport the fix to 3.10.

Reply:
Backports cleanly though. Thanks @tiran it's interesting rediscovering your work like this.